modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteSort.java

```diff
@@ -107,16 +128,24 @@ public IgniteSort(RelInput input) {
         return Pair.of(childTraits.replace(collation()), ImmutableList.of(childTraits));
     }
 
+    /** {@inheritDoc} */
+    @Override public double estimateRowCount(RelMetadataQuery mq) {
+        return memRows(super.estimateRowCount(mq));
```

Comment:
I think it's safer to use `mq.getRowCount(getInput())` but not rely on super method, since it can be overridden by `SortNode` at some point in time and will produce the wrong result.

Reply:
Fixed